tests/accessor/accessor_exceptions.hpp

```diff
@@ -0,0 +1,391 @@
+      "range of buffer in any dimension.");
+  SECTION(section_name) {
+    auto construct_acc = [&default_range, id](
+                             sycl::handler& cgh,
+                             sycl::buffer<DataT, Dimension> data_buf) {
+      sycl::accessor<DataT, Dimension, AccessMode, Target>(data_buf,
+                                                           default_range, id);
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to construct accessor from buffer, "
+      "tag, range and id. In case, when the sum of range and offset exceeds "
+      "the range of buffer in any dimension.");
+  SECTION(section_name) {
+    auto construct_acc = [&default_range, id](
+                             sycl::handler& cgh,
+                             sycl::buffer<DataT, Dimension> data_buf) {
+      sycl::accessor<DataT, Dimension>(data_buf, default_range, id,
+                                       sycl::read_only);
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to construct accessor from "
+      "buffer, handler and range. In case, the range exceeds the range of "
+      "buffer in any dimension");
+  SECTION(section_name) {
+    auto construct_acc = [&great_range](
+                             sycl::handler& cgh,
+                             sycl::buffer<DataT, Dimension> data_buf) {
+      sycl::accessor<DataT, Dimension, AccessMode, Target>(data_buf, cgh,
+                                                           great_range);
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to construct accessor from "
+      "buffer, handler, range and tag. In case, the range exceeds the range of "
+      "buffer in any dimension.");
+  SECTION(section_name) {
+    auto construct_acc = [&great_range](
+                             sycl::handler& cgh,
+                             sycl::buffer<DataT, Dimension> data_buf) {
+      sycl::accessor<DataT, Dimension>(data_buf, cgh, great_range,
+                                       sycl::read_only);
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to construct accessor from "
+      "buffer, handler, range and tag. In case, when the sum of range and "
+      "offset exceeds the range of buffer in any dimension.");
+  SECTION(section_name) {
```

Comment:
Actually, we can use it here, but I think we should avoid more that one nested `SECTION` (in this case), otherwise catch will generate too much extra runs

Reply:
What about `INFO` then?